packages/cli/src/UserManagement/routes/passwordReset.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable import/no-cycle */
+
+import express = require('express');
+import { v4 as uuidv4 } from 'uuid';
+import { URL } from 'url';
+import { genSaltSync, hashSync } from 'bcryptjs';
+
+import { Db, ResponseHelper } from '../..';
+import { N8nApp } from '../Interfaces';
+import { isValidEmail, validatePassword } from '../UserManagementHelper';
+import * as UserManagementMailer from '../email';
+import type { PasswordResetRequest } from '../Interfaces';
+import { issueJWT } from '../auth/jwt';
+
+export function addPasswordResetNamespace(this: N8nApp): void {
+	/**
+	 * Send a password reset email.
+	 */
+	this.app.post(
+		`/${this.restEndpoint}/forgot-password`,
+		ResponseHelper.send(async (req: PasswordResetRequest.Email) => {
+			const { email } = req.body;
+
+			if (!email) {
+				throw new ResponseHelper.ResponseError('Email is mandatory', undefined, 400);
+			}
+
+			if (!isValidEmail(email)) {
+				throw new ResponseHelper.ResponseError('Invalid email address', undefined, 400);
+			}
+
+			if (!req.headers.host) {
```

Comment:
why do we need the host in the headers?

Reply:
To create `baseUrl` on L49, to create the link emailed to the user.

Reply:
so we get the host from the FE? Shouldn't the backend know what it's host is?

Reply:
Found it, please check commit `473f133`